### PR TITLE
Add `epoch_bound` parameter to `RunningAverage`.

### PR DESCRIPTION
**Pull request by Evpok**
This optional parameter controls whether the running average should be reset after each epoch.

Check list:
* [ ] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/running_average.py [hunk 1]
```diff
@@ -67,8 +70,9 @@ def compute(self):
         return self._value
 
     def attach(self, engine, name):
-        # restart average every epoch
-        engine.add_event_handler(Events.EPOCH_STARTED, self.started)
+        if self.epoch_bound:
+            # restart average every epoch
+            engine.add_event_handler(Events.EPOCH_STARTED, self.started)
```
Comment (vfdev-5) [inline ignite/metrics/running_average.py]: I think we need to call started at least once. So you can define the event for `self.started` handlers as 
```python
event = Events.STARTED if self.epoch_bound else Events.EPOCH_STARTED
engine.add_event_handler(event, self.started)
```
What do you think ?

**Comment by vfdev-5** [comment]
@Evpok as discussed at slack, yes, I think one integration test is needed.

I also thinking about, shouldn't `epoch_bound` be an argument of `attach` instead of `__init__` ?

**Comment by Evpok** [comment]
Yes, it does make more sense

**Comment by Evpok** [comment]
Are other metrics with additional parameters to `attach` though @vfdev-5 ? Not sure how to document it :-)

**Comment by vfdev-5** [comment]
@Evpok thinking more on that, maybe we can leave `epoch_bound` argument in constructor such that it defines the way to compute a metric for any engine.

If you can add an integration test, it would awesome :)

**Comment by Evpok** [comment]
Right, working on it

**Comment by vfdev-5** [comment]
@Evpok could you please fix flake8 error

**Pull request merged**